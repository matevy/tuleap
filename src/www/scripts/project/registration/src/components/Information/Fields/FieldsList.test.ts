/*
 * Copyright (c) Enalean, 2019 - present. All Rights Reserved.
 *
 *  This file is a part of Tuleap.
 *
 * Tuleap is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * Tuleap is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Tuleap. If not, see <http://www.gnu.org/licenses/>.
 *
 */

import { shallowMount, Wrapper } from "@vue/test-utils";
import { createProjectRegistrationLocalVue } from "../../../helpers/local-vue-for-tests";
import FieldsList from "./FieldsList.vue";
import EventBus from "../../../helpers/event-bus";

describe("FieldsList - ", () => {
    let factory: Wrapper<FieldsList>;
    beforeEach(async () => {
        const field = {
            group_desc_id: "1",
            desc_name: "custom field",
            desc_type: "text",
            desc_description: "a helpful description",
            desc_required: true
        };

        factory = shallowMount(FieldsList, {
            localVue: await createProjectRegistrationLocalVue(),
            propsData: { field: field }
        });
    });

    it("Display correctly the component", () => {
        const wrapper = factory;

        expect(wrapper).toMatchSnapshot();
    });

    it("Send an event when user chooses a category", () => {
        const event_bus_emit = jest.spyOn(EventBus, "$emit");

        const wrapper = factory;

        wrapper.find("[data-test=project-field-text]").setValue("my new value");

        expect(event_bus_emit).toHaveBeenCalledWith("update-field-list", {
            field_id: "1",
            value: "my new value"
        });
    });
});
