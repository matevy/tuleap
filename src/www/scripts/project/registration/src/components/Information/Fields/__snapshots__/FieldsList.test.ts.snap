// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FieldsList -  Display correctly the component 1`] = `
<div class="tlp-form-element"><label for="input-1" class="tlp-label">
    custom field
    <!----></label> <textarea placeholder="a helpful description" id="textaarea-1" data-test="project-field-text" class="tlp-textarea"></textarea></div>
`;
